AsyncDisplayKit/Layout/ASLayoutOptionsPrivate.mm

```diff
@@ -0,0 +1,133 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import "ASLayoutOptionsPrivate.h"
+#import <AsyncDisplayKit/ASDisplayNodeInternal.h>
+
+
+#define ASLayoutOptionsForwarding \
```

Comment:
Very cute.  As an incremental improvement to readability, put a short comment above this calling attention to the strategy used here.  E.g. we are exposing a passthrough API on both ASDisplayNode and ASLayoutSpec for the convenience of the majority of developers, who will never need to customize ASLayoutOptions (and thus don't need to be aware of the object type's existence).  In order to avoid repetition, #defining the entire implementation allows it to be included in category extensions on both classes very easily.